### PR TITLE
Add ReplicateBlobRequest V2

### PR DESCRIPTION
**Pull request by JingQianCloud**
V2 added operationType, operationType, lifeVersion and one type specific parameter.

### CODE REVIEW

File: ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java [hunk 1]
```diff
@@ -21,18 +21,32 @@
 
 
 /**
+ * LOCAL_CONSISTENCY_TODO: Rename ReplicateBlobRequest to RepairBlobRequest
  * ReplicateBlob request to replicate one particular blob
  */
 public class ReplicateBlobRequest extends RequestOrResponse {
   public final static short VERSION_1 = 1;
+  public final static short VERSION_2 = 2;
   private final static short CURRENT_VERSION = VERSION_1;
```
Comment (SophieGuo410) [inline ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java:30]: nit: not used.

**Review by SophieGuo410 — APPROVED** [review]
lgtm! Just left some nit picking.

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java [hunk 2]
```diff
@@ -21,18 +21,32 @@
...
+
+  // the Blob to be repaired.
   private final BlobId blobId;
   // sourceHostName and sourceHostPort determine the source DataNodeId.
   private final String sourceHostName;
   private final int sourceHostPort;
 
+  // operation related parameters
+  private RequestOrResponseType operationType;
```
[the remark this answers is not in the document]
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java]: Added the "final"

File: ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java (hunk 1 above)
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java]: Removed.

File: ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java [hunk 3]
```diff
@@ -21,18 +21,32 @@
...
+  private long operationTimeInMs;
+  private short lifeVersion;
+  private long operationParameter; // It's the operation specific parameter. For TtlUpdate, it's the expirationTimeInMs.
```
[the remark this answers is not in the document]
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/ReplicateBlobRequest.java]: Renamed.

**Pull request merged**